File: lesson04/app/src/main/java/br/odb/nehe/lesson04/GL2JNILib.java
package br.odb.nehe.lesson04;

// Wrapper for native library

import android.content.res.AssetManager;

class GL2JNILib {

    public static AssetManager assetManager;

     static {
         System.loadLibrary("NdkGlue");
     }

    /**
     * @param width the current view width
     * @param height the current view height
     */
    public static native void init(int width, int height);
    public static native void step();
    public static native void tick();
    public static native void onDestroy();
    public static native void onCreate( AssetManager assetManager );
}

File: lesson04/app/src/main/java/br/odb/nehe/lesson04/GL2JNILib.kt
package br.odb.nehe.lesson04

import android.content.res.AssetManager

// Wrapper for native library
internal object GL2JNILib {

	/**
	 * @param width  the current view width
	 * @param height the current view height
	 */
	external fun init(width: Int, height: Int)
	external fun step()
	external fun tick()
	external fun onDestroy()
	external fun onCreate(assetManager: AssetManager?)

	init {
		System.loadLibrary("NdkGlue")
	}
}